lib/controllers/root-controller.js

```diff
@@ -800,3 +742,104 @@ export default class RootController extends React.Component {
     });
   }
 }
+
+class TabTracker {
+  constructor(component, name) {
+    this.component = component;
+    this.workspace = component.props.workspace;
+    this.name = name;
+
+    this.stateKey = `${name}TabActive`;
+    this.dockItemKey = `${name}DockItem`;
+    this.controllerKey = `${name}TabController`;
+  }
+
+  getState() {
+    return this.component.state[this.stateKey];
+  }
+
+  setStateKey(value) {
+    return new Promise(resolve => {
+      this.component.setState({[this.stateKey]: value}, resolve);
```

Comment:
Oh, I see...

... So, it turns out that JavaScript _does_ let you nest classes:

<img width="1016" alt="screen shot 2017-05-10 at 10 38 31 pm" src="https://cloud.githubusercontent.com/assets/17565/25929980/849bac84-35d1-11e7-8f8b-dbb3dcdc638b.png">

This might be a good candidate for that?